draft-ietf-lisp-pubsub.xml

```diff
@@ -303,13 +303,13 @@ Rec +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
 
-          <t>The subscription state can also be created explicitly by configuration at the Map-Server (possible when a pre-shared security association exists, see <xref target="security"></xref>) using a variety of means that are out of scope. If at the time the explicit subscription is configured there is no nonce that can be used for the explicit subscription state (e.g., from a different subscription already established with the same xTR when a single nonce is kept per xTR-ID), then both the xTR and Map-Server MUST be configured with the initial nonce to use. It is RECOMMENDED to have a configuration option to enable (or disable) the xTR to accept publication information for EID-Prefixes the xTR did not explicitly subscribe to. By default, the xTR is allowed to modify explicitly configured subscription state following the procedures described in this section, however the Map-Server MAY disable this via configuration. If the Map-Server is instructed to not allow xTRs to modify explicitly configured subscriptions, whenever an xTR tries to modify an explicitly configured subscription the Map-Server SHOULD return a Negative Map-Reply (Loc-Count = 0 and Loc-AFI = 0) with ACT bits set to 4 "Drop/Policy-Denied". The Map-Server MAY silently ignore these requests if, for example, they are received within a short interval.</t>
+          <t>The subscription state can also be created explicitly by configuration at the Map-Server (possible when a pre-shared security association exists, see <xref target="security"></xref>) using a variety of means that are out of scope. If at the time the explicit subscription is configured there is no nonce that can be used for the explicit subscription state (e.g., from a different subscription already established with the same xTR when a single nonce is kept per xTR-ID), then both the xTR and Map-Server MUST be configured with the initial nonce to use. It is RECOMMENDED to have a configuration option to enable (or disable) the xTR to accept publication information for EID-Prefixes the xTR did not explicitly subscribe to. By default, the xTR is allowed to modify explicitly configured subscription state following the procedures described in this section, however this MAY be disabled at the Map-Server via configuration. If the Map-Server is instructed to not allow xTRs to modify explicitly configured subscriptions, and an xTR tries to do so, the Map-Server will follow the usual process described in this section to handle failed subscription requests (i.e. responding with a Map-Reply).</t>
```

Comment:
```suggestion
          <t>The subscription state can also be created explicitly by configuration at the Map-Server (possible when a pre-shared security association exists, see <xref target="security"></xref>) using a variety of means that are out of scope. If at the time the explicit subscription is configured there is no nonce that can be used for the explicit subscription state (e.g., from a different subscription already established with the same xTR when a single nonce is kept per xTR-ID), then both the xTR and Map-Server MUST be configured with the initial nonce to use. It is RECOMMENDED to have a configuration option to enable (or disable) the xTR to accept publication information for EID-Prefixes the xTR did not explicitly subscribe to. By default, the xTR is allowed to modify explicitly configured subscription state following the procedures described in this section, however this MAY be disabled at the Map-Server via configuration. If the Map-Server is instructed to not allow xTRs to modify explicitly configured subscriptions, and an xTR tries to do so, the Map-Server follows the procedure described in this section for handling failed subscription requests (i.e., responding with a Negative Map-Reply).</t>
```

Reply:
this should be a regular Map-Reply, not a negative one

Reply:
I think this one should be negative Map-Reply as the TR as requesting an existing subscription.

Reply:
In general the behavior is that if the MS doesn't accept a subscription request (for whichever reason), it should reply to the xTR with a regular Map-Reply. An unsubscription request should be no different IMO. I'm concerned that if we reply with a Negative Map-Reply we might delete a perfectly valid map-cache entry on the xTR when there's no reason to do so.

Reply:
Since this is the last remaining open item, I'm merging this as is @boucadair, in order to generate version -14. We can continue the conversation on this after and make small changes if needed.

Reply:
The issue with a (positive) Map-Reply is that the xTR does not know how to interpret it. The worse case is that it can interpret this as a subscription failure, and the reissue a subscription request or send regular refresh requests.

An example of an update I have in mind is updating the RLOCs (this can fail because of the policy related to explicit subs, the xTR provided an RLOC list that exceeds an allowed max). Returning a normal Map-Reply can be seen as if the subscription as a whole was removed, not that this is only about this update. The behavior is not deterministic with the current text in the draft. 

With the negative Map-Reply, we have an explicit signal about the reason of the failure without touching any existing cache. The text can be clarified to indicate that the cache is not touched when the xTR asked for an update of the subscription state, but that subscription failed.

Reply:
A returned Map-Reply doesn't mean the subscription was removed, it means that the subscription request was not accepted. Returning a Negative Map-Reply has the same implications, just that in that case the xTR assumes subscription not accepted and no RLOCs available for that EID. 

If we want to add explicit machinery for this we need to go back to the text we had that used NMR w/ ACT = 4 and then add the text saying that the map-cache is not updated in that case.

Reply:
> If we want to add explicit machinery for this we need to go back to the text we had that used NMR w/ ACT = 4 and then add the text saying that the map-cache is not updated in that case.

I prefer this. Thanks